From 31616ab809fcdd147e516c8b634324fa023f2c5b Mon Sep 17 00:00:00 2001
From: Nicola Fontana <ntd@entidi.it>
Date: Mon, 23 Mar 2015 19:15:11 +0100
Subject: [PATCH 1/6] Remove leading $SHELL in libtool call

Avoid crashes due to double shell in subprocess.Popen() on linux, e.g.:

>>> import subprocess
>>> subprocess.Popen(['/bin/sh', '/bin/sh'])
/bin/sh: /bin/sh: cannot execute binary file

https://bugzilla.gnome.org/show_bug.cgi?id=761982
---
 giscanner/ccompiler.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/giscanner/ccompiler.py b/giscanner/ccompiler.py
index 83fc789..6ddbdfe 100644
--- a/giscanner/ccompiler.py
+++ b/giscanner/ccompiler.py
@@ -277,7 +277,8 @@ class CCompiler(object):
         else:
             libtool = utils.get_libtool_command(options)
             if libtool:
-                args.append(utils.which(os.environ.get('SHELL', 'sh.exe')))
+                if os.name == 'nt':
+                    args.append(utils.which(os.environ.get('SHELL', 'sh.exe')))
                 args.extend(libtool)
                 args.append('--mode=execute')
             args.extend([os.environ.get('DLLTOOL', 'dlltool.exe'), '--identify'])
-- 
2.7.1

